### PR TITLE
qa: Enforce JSON linter for GCT configs

### PR DESCRIPTION
**Pull request by thrasher-**
# PR Description

Adds a new GHA and Makefile utility command to enforce JSON format standards for our configs using jq. Also sorts the exchanges by name for now rather than the whole file for starters.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How has this been tested

- [x] go test ./... -race
- [x] golangci-lint run
- [x] GHA
- [x] `make lint_configs` and manual breaking to make sure the diff wigs out

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Github Actions with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/configs-json-lint.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+jobs:
+  lint:
+    name: configs JSON lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Check configs JSON format
+        run: |
+          files=("config_example.json" "testdata/configtest.json")
```
Comment (shazbert) [inline .github/workflows/configs-json-lint.yml:13]: Suggestion: might be able to use `files=$(git ls-files '*.json')` so we can conform everything and for future files and filter by config key word. But that seems like a pain. So only a suggestion. 

here's the output:

image

**Review by shazbert — APPROVED** [review]
Tickity Tack.

**Review by gloriousCode — APPROVED** [review]
tACK!
```
make lint_configs
Checking if jq is installed...
Processing config_example.json...
jq: parse error: Invalid numeric literal at line 16, column 0
jq processing failed on *.json
make: *** [lint_configs] Error 1

make lint_configs
Checking if jq is installed...
Processing config_example.json...
Processing testdata/configtest.json...
```

Confirmed that it works 🌞  Emoji's inside yml files is approved too

**Review by gloriousCode — APPROVED** [review]
retACK! Increase in actual human friendliness detected
```
make lint_configs
-n Checking if jq is installed... 
OK
-n Processing config_example.json... 
OK
-n Processing testdata/configtest.json... 
OK

make lint_configs
-n Checking if jq is installed... 
OK
-n Processing config_example.json... 
OK
-n Processing testdata/configtest.json... 
jq: parse error: Invalid numeric literal at line 462, column 0
FAILED
make: *** [lint_configs] Error 1

```

File: .github/workflows/configs-json-lint.yml (hunk 1 above)
Comment (thrasher-) [reply .github/workflows/configs-json-lint.yml:13]: Yep, it was made to be extensible but this PR only targets our mainly touched configs for now

**Review by shazbert — APPROVED** [review]
ch tack

**Review by gloriousCode — APPROVED** [review]
```
make lint_configs
Checking if jq is installed... OK
Processing config_example.json... OK
Processing testdata/configtest.json... OK
```

```
make lint_configs
Checking if jq is installed... OK
Processing config_example.json... OK
Processing testdata/configtest.json... jq: parse error: Invalid numeric literal at line 462, column 0
FAILED
make: *** [lint_configs] Error 1
```
Now with less -n

**Pull request merged**